test/node/medtronic/testMedtronicSimulator.js

```diff
@@ -0,0 +1,669 @@
+        delete basal1.jsDate;
+
+        expect(simulator.getEvents()).deep.equals([
+          settings,
+          basal1.done(),
+          expectedTempBasal1,
+          expectedTempBasal2
+        ]);
+      });
+
+      it('checks for schedule change over midnight', function() {
+
+        simulator.pumpSettings(settings);
+        simulator.basal(basal1);
+        simulator.basal(tempBasalOverMidnight);
+        simulator.basal(basal2);
+
+        var expectedTempBasal1 = _.cloneDeep(tempBasalOverMidnight.done());
+        expectedTempBasal1.suppressed.rate = 1.3;
+        expectedTempBasal1.duration = 3000000;
+        expectedTempBasal1.payload.duration = 7200000;
+        delete expectedTempBasal1.index;
+        delete expectedTempBasal1.jsDate;
+
+        var expectedTempBasal2 = _.cloneDeep(expectedTempBasal1);
+        expectedTempBasal2.duration = 4200000;
+        expectedTempBasal2.time = '2014-09-25T23:00:00.000Z';
+        expectedTempBasal2.deviceTime = '2014-09-26T00:00:00';
+        expectedTempBasal2.annotations = [{code: 'medtronic/basal/fabricated-from-schedule'}];
+        expectedTempBasal2.suppressed.rate = 0.2;
+        delete expectedTempBasal2.payload.duration;
+
+        delete basal1.index;
+        delete basal1.jsDate;
+
+        expect(simulator.getEvents()).deep.equals([
+          settings,
+          basal1.done(),
+          expectedTempBasal1,
+          expectedTempBasal2
+        ]);
+      });
+
+      it('checks temp basal schedule change that only happens after midnight', function() {
+
+        settings.basalSchedules.standard[0].rate = 1.3; // scheduled rate does not change at midnight
+        settings.basalSchedules.standard[1].start = 3600000; // schedule only changes at 1am
+
+        simulator.pumpSettings(settings);
+        simulator.basal(basal1);
+        simulator.basal(tempBasalOverMidnight);
+        simulator.basal(basal2);
+
+        var expectedTempBasal1 = _.cloneDeep(tempBasalOverMidnight.done());
+        expectedTempBasal1.suppressed.rate = 1.3;
+        expectedTempBasal1.duration = 6600000;
+        delete expectedTempBasal1.index;
+        delete expectedTempBasal1.jsDate;
+
+        var expectedTempBasal2 = _.cloneDeep(expectedTempBasal1);
+        expectedTempBasal2.duration = 600000;
+        expectedTempBasal2.time = '2014-09-26T00:00:00.000Z';
+        expectedTempBasal2.deviceTime = '2014-09-26T01:00:00';
+        expectedTempBasal2.annotations = [{code: 'medtronic/basal/fabricated-from-schedule'}];
+        expectedTempBasal2.suppressed.rate = 0.375;
+        delete expectedTempBasal2.payload.duration;
+
+        delete basal1.index;
+        delete basal1.jsDate;
+
+        expect(simulator.getEvents()).deep.equals([
+          settings,
+          basal1.done(),
+          expectedTempBasal1,
+          expectedTempBasal2
+        ]);
+      });
+
+      it('restarts temp basal after resume, with schedule change during suspend', function() {
```

Comment:
Maybe add another test case re: *not* resuming temp basal if it would have run its course completely by the time of the resumption?